src/fs/path.rs

```diff
@@ -0,0 +1,240 @@
+//! Module for [Path].
+
+use core::fmt::{Display, Formatter};
+
+/// Separator used for paths accross multiple directories. Also used as identifier for the
+/// root path.
+pub const SEPARATOR: char = '/';
+/// [SEPARATOR] but as string.
+const SEPARATOR_STR: &str = "/";
```

Comment:
Using `/` instead of `\` as the separator feels a little surprising. I recognize though that it is generally more convenient to write paths with `/` so that escaping (or prefixing the string with `r`) aren't needed. Maybe it would be better to allow both `/` and `\`? I think that's the [behavior of `std::path::Path` on Windows](https://doc.rust-lang.org/std/path/struct.Path.html).

Reply:
I decided to use one single option, i.e., don't provide support for both, for simplicity. I do not insist on it.

Reply:
I think if we go with a single separator I would prefer `\`. I think that will reduce the potential for confusion since we can just say "everything uses `\`", and it's helpful for any interactions between this API and other parts of the API that use paths with `\` (e.g. [`FilePathMediaDevicePath`](https://docs.rs/uefi/latest/uefi/proto/device_path/struct.FilePathMediaDevicePath.html))